geode-lucene/src/test/java/org/apache/geode/cache/lucene/FlatFormatSerializerJUnitTest.java

```diff
@@ -31,29 +38,75 @@
 
   @Test
   public void shouldParseMultiLayerNestObject() {
-    String[] fields = new String[] {"name", "contact.name", "contact.email", "contact.revenue",
-        "contact.address", "contact.homepage.id", "contact.homepage.title",
-        "contact.homepage.content", LuceneService.REGION_VALUE_FIELD};
+    String[] fields = new String[] {"name", "phoneNumbers", "myHomePages.content", "contacts.name",
```

Comment:
I think this test needs to be broken up into smaller test methods that each test a single thing.